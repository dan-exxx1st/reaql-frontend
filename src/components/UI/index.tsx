export { default as Avatar } from './Avatar';
export { default as Button } from './Button';
export { default as Circle } from './Circle';
export { default as SkeletonLoader } from './SkeletonLoader';
export { default as TextField } from './TextField';
export { default as Typography } from './Typography';
export { default as Wrapper } from './Wrapper';
export { default as CheckMark } from './CheckMark';
export { default as Icon } from './Icon';
export { default as CheckBox } from './CheckBox';
export { default as SystemMessage } from './SystemMessage';
export { default as Spinner } from './Spinner';
