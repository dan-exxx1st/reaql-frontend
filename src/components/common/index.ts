export { default as DialogHeader } from './DialogHeader';
export { default as ContactItem } from './ContactsItem';
export { default as UserHeader } from './UserHeader';
export { default as DialogTextField } from './DialogTextField';
export { default as MessageItem } from './MessageItem';
export { default as DialogHeaderBackdrop } from './DialogHeaderBackdrop';
export { default as MessageSearch } from './MessageSearch';
