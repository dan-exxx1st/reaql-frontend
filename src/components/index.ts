export { default as DialogListItem } from './common/DialogListItem';
export { default as EmojiPicker } from './common/EmojiPicker';
