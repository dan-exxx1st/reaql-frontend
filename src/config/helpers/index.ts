import { useState, useEffect } from 'react';

export function useLoaded({ src }: { src?: HTMLImageElement['src'] }) {
	const [loaded, setLoaded] = useState('false');

	useEffect(() => {
		if (!src) {
			return undefined;
		}

		setLoaded('loading');

		let active = true;
		const image = new Image();
		image.src = src;
		image.onload = () => {
			if (!active) {
				return;
			}
			setLoaded('loaded');
		};
		image.onerror = () => {
			if (!active) {
				return;
			}
			setLoaded('error');
		};

		return () => {
			active = false;
		};
	}, [src]);

	return loaded;
}
