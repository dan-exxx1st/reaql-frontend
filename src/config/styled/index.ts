import colorTheme from './colors';
import fontTheme from './fonts';

export { colorTheme, fontTheme };

export default {
    ...colorTheme,
    ...fontTheme,
};
