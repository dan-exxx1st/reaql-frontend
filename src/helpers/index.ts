import { parse } from 'querystring';

export function getDialogIdFromSearch(search: string) {
    const params = parse(search);

    const dialogId = Object.entries(params)
        .map(([key, value]) => key.indexOf('dialog') !== -1 && value)
        .toString();

    return dialogId;
}
