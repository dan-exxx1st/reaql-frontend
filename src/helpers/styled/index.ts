import colorTheme from './colors';
import fontTheme from './fonts';

export { colorTheme, fontTheme };

const effects = {
    formBoxShadow: '5px 5px 15px rgba(37, 37, 37, 0.1)',
};

export default {
    ...colorTheme,
    ...fontTheme,
    ...effects,
};
