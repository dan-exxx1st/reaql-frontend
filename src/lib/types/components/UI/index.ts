import { MouseEventHandler } from 'react';
import Theme from 'helpers/styled';
import { ColorType, IconName } from 'lib/types';
import { TypographyVariantType, TypographyComponentType } from 'lib/types/ui';

export interface IAvatarProps {
    src?: HTMLImageElement['src'];
    alt?: string;
    className?: string;
}

export interface IButtonProps {
    className?: string;
    theme: typeof Theme;
    onClick?: MouseEventHandler;
    variant: 'filled' | 'outlined';
    color: ColorType;
}

export interface ICheckMarkProps {
    isChecked: boolean;
    isDouble: boolean;
}

export interface ICircleProps {
    color: ColorType;
    iconName?: IconName;
    text?: string | number;
    width?: string;
    height?: string;
    cursor?: string;
    className?: string;
    onClick?: () => void;
}

export interface IIconProps {
    iconName: IconName;
    className?: string;
}

export interface ISkeletonLoaderProps {
    variant?: 'avatar' | 'block';
    width?: string;
    height?: string;
    className?: string;
}

export interface ITextFieldProps {
    onChange?: (value: string) => void;
    value?: string;
    className?: string;
    fieldSize: 'small' | 'medium' | 'large';
    color: ColorType;
    icon?: IconName;
    width?: string;
    height?: string;
    wrapperWidth?: string;
    placeholder?: string;
    type?: string;
}

export interface ITypographyProps {
    className?: string;
    variant: TypographyVariantType;
    component?: TypographyComponentType;
    color?: string;
    theme: typeof Theme;
}

export type { IWrapperProps } from './WrapperTypes';

export interface ICheckBoxProps {
    onClick?: () => void;
    text?: string;
}
