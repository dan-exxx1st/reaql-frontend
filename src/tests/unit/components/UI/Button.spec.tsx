import React from 'react';
import { Button } from 'components/UI';
import { colorTheme } from 'config/styled';
import { mountWithTheme } from 'tests/helpers/withTheme';

describe('<Button />', () => {
    describe('Snapshots', () => {
        it('Primary filled button without effects', () => {
            const wrapper = mountWithTheme(
                <Button variant="filled" color="primary">
                    button
                </Button>
            );
            expect(wrapper).toMatchSnapshot();
            expect(wrapper).toHaveStyleRule('background', colorTheme.pc);
            expect(wrapper).toHaveStyleRule('color', colorTheme.wc);
        });

        it('Primary outlined button without effects', () => {
            const wrapper = mountWithTheme(
                <Button variant="outlined" color="primary">
                    button
                </Button>
            );
            expect(wrapper).toMatchSnapshot();
            expect(wrapper).toHaveStyleRule('background', 'none');
            expect(wrapper).toHaveStyleRule('color', colorTheme.pc);
            expect(wrapper).toHaveStyleRule(
                'border',
                `1px solid ${colorTheme.pc}`
            );
        });

        it('Secondary filled button without effects', () => {
            const wrapper = mountWithTheme(
                <Button variant="filled" color="secondary">
                    button
                </Button>
            );
            expect(wrapper).toMatchSnapshot();
            expect(wrapper).toHaveStyleRule('background', colorTheme.sc);
            expect(wrapper).toHaveStyleRule('color', colorTheme.dgc);
            expect(wrapper).toHaveStyleRule('border', 'none');
        });

        it('Secondary outlined button without effects', () => {
            const wrapper = mountWithTheme(
                <Button variant="outlined" color="secondary">
                    button
                </Button>
            );

            expect(wrapper).toHaveStyleRule('background', 'none');
            expect(wrapper).toHaveStyleRule(
                'border',
                `1px solid ${colorTheme.sc}`
            );
            expect(wrapper).toHaveStyleRule('color', colorTheme.dgc);
            expect(wrapper).toMatchSnapshot();
        });
    });

    it('Button should render text which passed to props', () => {
        const wrapper = mountWithTheme(
            <Button variant="filled" color="primary">
                Default button
            </Button>
        );
        expect(wrapper).toHaveStyleRule('background', colorTheme.pc);
    });

    it('Button should use onClick event', () => {
        const mockFunction = jest.fn();
        const wrapper = mountWithTheme(
            <Button
                variant="filled"
                color="primary"
                onClick={mockFunction}
                className="custom-class"
            >
                Should click
            </Button>
        );
        wrapper.simulate('click');
        expect(mockFunction).toBeCalled();
    });
});
