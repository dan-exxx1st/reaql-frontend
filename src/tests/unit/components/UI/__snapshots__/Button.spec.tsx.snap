// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Snapshots Primary filled button to match snapshot without effects 1`] = `
.c1 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 14px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  margin: 0;
}

.c0 {
  border-radius: 5px;
  outline: none;
  border: none;
  font-family: 'Roboto',sans-serif;
  padding: 7px 26px;
  cursor: pointer;
  max-height: 50px;
  background: #00ADFF;
  color: #FAFAFA;
}

.c0:hover {
  box-shadow: 1px 1px 10px rgb(10,10,10,0.25);
}

<button
  className="c0"
>
  <p
    className="c1"
  >
    To match snapshot button
  </p>
</button>
`;

exports[`<Button /> Snapshots Primary outlined button to match snapshot without effects 1`] = `
.c1 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 14px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  margin: 0;
}

.c0 {
  border-radius: 5px;
  outline: none;
  border: none;
  font-family: 'Roboto',sans-serif;
  padding: 7px 26px;
  cursor: pointer;
  max-height: 50px;
  border: 1px solid #00ADFF;
  background: none;
  color: #00ADFF;
}

.c0:hover {
  box-shadow: 1px 1px 10px rgb(10,10,10,0.25);
}

<button
  className="c0"
>
  <p
    className="c1"
  >
    To match snapshot button
  </p>
</button>
`;

exports[`<Button /> Snapshots Secondary filled button to match snapshot without effects 1`] = `
.c1 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 14px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  margin: 0;
}

.c0 {
  border-radius: 5px;
  outline: none;
  border: none;
  font-family: 'Roboto',sans-serif;
  padding: 7px 26px;
  cursor: pointer;
  max-height: 50px;
  background: #E6E6E6;
  color: #A5A5A5;
}

.c0:hover {
  box-shadow: 1px 1px 10px rgb(10,10,10,0.1);
}

<button
  className="c0"
>
  <p
    className="c1"
  >
    To match snapshot button
  </p>
</button>
`;

exports[`<Button /> Snapshots Secondary outlined button to match snapshot without effects 1`] = `
.c1 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 14px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  margin: 0;
}

.c0 {
  border-radius: 5px;
  outline: none;
  border: none;
  font-family: 'Roboto',sans-serif;
  padding: 7px 26px;
  cursor: pointer;
  max-height: 50px;
  border: 1px solid #E6E6E6;
  color: #A5A5A5;
  background: none;
}

.c0:hover {
  box-shadow: 1px 1px 10px rgb(10,10,10,0.1);
}

<button
  className="c0"
>
  <p
    className="c1"
  >
    To match snapshot button
  </p>
</button>
`;
